Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -435,30 +467,14 @@ open class RouteController: NSObject {
         BillingHandler.shared.beginBillingSession(for: .activeGuidance, uuid: sessionUUID)
 
         subscribeNotifications()
-        updateNavigator(with: routeProgress)
-        updateObservation(for: routeProgress)
+        updateNavigator(with: routeProgress, completion: {_ in })
     }
     
     deinit {
         BillingHandler.shared.stopBillingSession(with: sessionUUID)
-        
-        resetObservation(for: routeProgress)
         unsubscribeNotifications()
     }
-    
-    func resetObservation(for progress: RouteProgress) {
-        progress.legIndexHandler = nil
-    }
-    
-    func updateObservation(for progress: RouteProgress) {
-        progress.legIndexHandler = { [weak self] (oldValue, newValue) in
-            guard newValue != oldValue else {
-                return
-            }
-            self?.updateRouteLeg(to: newValue)
```

Comment:
You are not supposed to change legIndex manually in progress. It should only change in response to Navigator status update.